src/Ocelot/DependencyInjection/OcelotBuilder.cs

```diff
@@ -82,6 +84,10 @@ public OcelotBuilder(IServiceCollection services, IConfiguration configurationRo
             Services.TryAddSingleton<IFileConfigurationRepository, DiskFileConfigurationRepository>();
             Services.TryAddSingleton<IFileConfigurationSetter, FileAndInternalConfigurationSetter>();
             Services.TryAddSingleton<IServiceDiscoveryProviderFactory, ServiceDiscoveryProviderFactory>();
+            Services.AddSingleton<ILoadBalancerCreator, NoLoadBalancerCreator>();
```

Comment:
Why not TryAddIngleton ?

Reply:
Because ```LoadBalancerFactory``` expects an ```IEnumerable<ILoadBalancerCreator>```, not a single one.
When registering ```ILoadBalancerCreator``` implementations using ```TryAddSingleton```, only the first one in ```OcelotBuilder``` would be registered.

See unit test ```should_add_custom_load_balancer_creators_by_di_factory_method_with_arguments```.